projects/packages/paypal-payments/src/block/simple-payments.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Pay with PayPal block (aka Simple Payments).
+ *
+ * @since 9.0.0
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\PaypalPayments\SimplePayments;
+
+use Automattic\Jetpack\Blocks;
+use Jetpack_Simple_Payments;
+use WP_Post;
+
+const FEATURE_NAME = 'simple-payments';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	Blocks::jetpack_register_block(
+		__DIR__,
+		array(
+			'render_callback' => __NAMESPACE__ . '\render_block',
+			'plan_check'      => true,
+		)
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Pay with PayPal block dynamic rendering.
+ *
+ * @param array  $attr    Array containing the block attributes.
+ * @param string $content String containing the block content.
+ *
+ * @return string
+ */
+function render_block( $attr, $content ) {
+	// Do nothing if block content is a `simple-payment` shortcode.
+	if ( preg_match( '/\[simple-payment(.*)]/', $content ) ) {
+		return $content;
+	}
+
+	// Keep content as-is if rendered in other contexts than frontend (i.e. feed, emails, API, etc.).
+	if ( ! jetpack_is_frontend() ) {
```

Comment:
This will cause issues once we get rid of the Jetpack. I would recommend adding a `function_exists` check right now to save us some surprises later.

Reply:
Good catch! Added in [b1391f3](https://github.com/Automattic/jetpack/pull/43413/commits/b1391f3187e7307420c0a1ecbda00c3b2d01bb25).